include/rapidjson/prettywriter.h

```diff
@@ -184,8 +186,16 @@ class PrettyWriter : public Writer<OutputStream, SourceEncoding, TargetEncoding,
     //@{
 
     //! Simpler but slower overload.
-    bool String(const Ch* str) { return String(str, internal::StrLen(str)); }
-    bool Key(const Ch* str) { return Key(str, internal::StrLen(str)); }
+    template <typename T>
+    bool String(const T* const& str, RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return String(str, internal::StrLen(str)); }
+    template <typename T>
+    bool Key(const T* const& str, RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return Key(str, internal::StrLen(str)); }
+    
+    //! The compiler can give us the length of quoted strings for free.
+    template <typename T, size_t N>
+    bool String(const T (&str)[N], RAPIDJSON_ENABLEIF((internal::IsSame<Ch, T>))) { return String(str, N-1); }
```

Comment:
This overload makes string "abc" in the following case:
```
char abcd[] = {'a', 'b', 'c', 'd'};
writer.String(abcd);
```
This behaviour is unexpected, and that's why I generally avoid this trick in favor of strlen.